chain/network/src/peer_manager.rs

```diff
@@ -268,25 +268,28 @@ impl PeerManagerActor {
 
     /// Returns single random peer with the most weight.
     fn most_weight_peers(&self) -> Vec<FullPeerInfo> {
-        let max_weight = match self
+        // This finds max of weight and height and returns such height.
+        let height_with_max_weight = match self
             .active_peers
             .values()
-            .map(|active_peer| active_peer.full_peer_info.chain_info.total_weight)
+            .map(|active_peers| {
+                (
+                    active_peers.full_peer_info.chain_info.total_weight,
+                    active_peers.full_peer_info.chain_info.height,
+                )
+            })
             .max()
         {
-            Some(w) => w,
-            None => {
-                return vec![];
-            }
+            Some((_, height)) => height,
```

Comment:
So out of all peers with the largest total_weight it gets the one with the largest height and returns it, correct?